### PR TITLE
CMR-7710 - Read DOIs from ISO records from the citation identifier using Anchor tag using the xlink/title attribute = DOI

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj [hunk 1]
```diff
@@ -16,13 +16,24 @@
   "This is used to check if the constant DOI is contained in the identification description element."
   "DOI")
 
+(defn- is-doi-identified-by-attribute?
+  "Checks to see if the Anchor contains an <gmx:Anchor xlink:title=\"DOI\" attribute. If it does then the
+   identifier is a DOI. Returns true if sucessful or nil if not."
+  [gmd-id]
+  (let [attrs (get (first (select gmd-id "gmd:MD_Identifier/gmd:code/gmx:Anchor")) :attrs)]
+    (some #(when (= :xlink/title (key %))
+             (when (= "DOI" (val %))
```
Comment (eereiter) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj:25]: @sxu123 sxu123 37 minutes ago Member
Could you change it to (when (and (= :xlink/title (key %) (= "DOI" (val %))?

@ygliuvt ygliuvt 24 minutes ago Member
No. some takes a function.

**Review by ygliuvt — APPROVED** [review]

**Review by sxu123 — APPROVED** [review]

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj (hunk 1 above)
Comment (sxu123) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj:25]: Right. how about (and  (= :xlink/title (key %)) (= "DOI" (val %)))

Comment (ygliuvt) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj:25]: No. `#` is clojure's way to start an anonymous function. Without `#`, `%` is meaningless.

Comment (sxu123) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj:25]: with # of course.

Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj:25]: I can change the code to look like the following:
    (some #(when (and (= :xlink/title (key %))
                                      (= "DOI" (val %)))
                      true)
               attrs)))

Comment (sxu123) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj:25]: I just tested it.

Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj]: I have updated the code to use (some #(and (= :xlink/title (key %) (= "DOI" (val %))) attrs)

**Pull request merged**